pkg/keymanagementprovider/azurekeyvault/provider_test.go

```diff
@@ -185,19 +204,81 @@ func (m *MockKeyKVClient) GetKey(ctx context.Context, keyName string, keyVersion
 	}
 	return azkeys.GetKeyResponse{}, nil
 }
+
+func (m *MockKeyKVClient) NewListKeyVersionsPager(keyName string, options *azkeys.ListKeyVersionsOptions) *runtime.Pager[azkeys.ListKeyVersionsResponse] {
+	if m.NewListKeyVersionsPagerFunc != nil {
+		return m.NewListKeyVersionsPagerFunc(keyName, options)
+	}
+	KeyCreated := time.Now()
+	return runtime.NewPager(runtime.PagingHandler[azkeys.ListKeyVersionsResponse]{
+		More: func(_ azkeys.ListKeyVersionsResponse) bool {
+			return false
+		},
+		Fetcher: func(_ context.Context, _ *azkeys.ListKeyVersionsResponse) (azkeys.ListKeyVersionsResponse, error) {
+			var resp azkeys.ListKeyVersionsResponse
+			var keyID azkeys.ID = "https://testkv.vault.azure.net/keys/key1/c1f03df1113d460491d970737dfdc35d"
```

Comment:
General comment not specific to this PR: I was reading on some security best practices and it's recommended not to use valid resource IDs or http urls that you don't own unless it's absolutely necessary to testing the functionality

Reply:
Good to know. Thank you for sharing your learnings. :)